components/ui/Dialog.tsx

```diff
@@ -0,0 +1,250 @@
+'use client';
+
+import * as DialogPrimitive from '@radix-ui/react-dialog';
+import { motion } from 'framer-motion';
+import {
+  BookmarkCheck,
+  ChevronDown,
+  Info,
+  OctagonAlert,
+  TriangleAlert,
+  X,
+} from 'lucide-react';
+import * as React from 'react';
+import { tv } from 'tailwind-variants';
+import {
+  DialogVariants,
+  type Dialog as DialogType,
+  type DialogVariant,
+} from '~/lib/dialogs/dialog-schemas';
+import { Button } from './Button';
+
+// following this docs to implement this
+// https://www.tailwind-variants.org/docs/slots#slots-with-variants
+const dialogElements = tv({
+  slots: {
+    overlay: 'fixed inset-0 z-50 bg-rich-black/50',
+    content:
+      'fixed left-[50%] top-[50%] z-50 grid w-full max-w-sm sm:max-w-lg gap-1 border bg-background p-5 shadow-lg duration-200 rounded-md sm:rounded-lg',
+    close:
+      'focus:ring-ring absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground',
+    title: 'flex items-center gap-1.5 text-xl font-bold',
+    contentMain: 'text-foreground',
+    footer: 'flex flex-col-reverse sm:flex-row sm:justify-end gap-2',
```

Comment:
Youre using some of the tailwind default theme classes for styling (text-muted for example), but we won't be using it. Its better to keep to very basic classes for now.